homeassistant/components/reolink/host.py

```diff
@@ -237,25 +237,35 @@ async def async_init(self) -> None:
                     self._async_check_onvif_long_poll,
                 )
 
-        if self._api.sw_version_update_required:
-            ir.async_create_issue(
-                self._hass,
-                DOMAIN,
-                "firmware_update",
-                is_fixable=False,
-                severity=ir.IssueSeverity.WARNING,
-                translation_key="firmware_update",
-                translation_placeholders={
-                    "required_firmware": self._api.sw_version_required.version_string,
-                    "current_firmware": self._api.sw_version,
-                    "model": self._api.model,
-                    "hw_version": self._api.hardware_version,
-                    "name": self._api.nvr_name,
-                    "download_link": "https://reolink.com/download-center/",
-                },
-            )
-        else:
-            ir.async_delete_issue(self._hass, DOMAIN, "firmware_update")
+        ch_list: list[int | None] = [None]
+        if self._api.is_nvr:
+            ch_list.extend(self._api.channels)
+        for ch in ch_list:
+            if not self._api.supported(ch, "firmware"):
+                continue
+
+            key = ch if ch is not None else "host"
+            if self._api.camera_sw_version_update_required(ch):
+                ir.async_create_issue(
+                    self._hass,
+                    DOMAIN,
+                    f"firmware_update_{key}",
```

Comment:
Should this now create 1 issue overall or 1 for every device that has this?

Reply:
1 issue per device that is running a incompatible firmware version.
The issue title will have the name of that specific camera/device in it.

This is because if someone has mulitple identical camera's they can still be running diffrent firmware versions (if they forgot to update 1 of them or had one laying around in a box and install later).